redis-lettuce/src/main/java/io/micronaut/configuration/lettuce/DefaultRedisClusterClientFactory.java

```diff
@@ -40,45 +37,30 @@
 @Requires(property = RedisSetting.REDIS_URIS)
 @Singleton
 @Factory
-public class DefaultRedisClusterClientFactory {
+public class DefaultRedisClusterClientFactory extends AbstractRedisClusterClientFactory {
 
-    /**
-     * Create the client based on config URIs.
-     * @param config config
-     * @param defaultClientResources default {@link ClientResources}
-     * @return client
-     */
+    @Override
     @Bean(preDestroy = "shutdown")
     @Singleton
     @Primary
-    public RedisClusterClient redisClient(@Primary AbstractRedisConfiguration config, @Primary @Nullable ClientResources defaultClientResources) {
-        List<RedisURI> uris = config.getUris();
-        if (CollectionUtils.isEmpty(uris)) {
-            throw new ConfigurationException("Redis URIs must be specified");
-        }
-        return defaultClientResources == null ? RedisClusterClient.create(uris) : RedisClusterClient.create(defaultClientResources, uris);
+    public RedisClusterClient redisClient(@Primary AbstractRedisConfiguration config,
```

Comment:
Deleting the old constructor would be a breaking change. You will need to deprecated it.

Reply:
Actually, I didn't delete the old constructor, I implemented it in [AbstractRedisClusterClientFactory.java](https://github.com/micronaut-projects/micronaut-redis/pull/244/files/6173905cb7d7b935eec544650d86feece2cdd1c7?diff=split&w=1#diff-9e98db04d68a3d2e9810c1652bd1a4a4a0c92339b293a94b3f501a417284cc5b), (lines 41-44) as deprecated function.

Reply:
This class extends [AbstractRedisClusterClientFactory.java](https://github.com/micronaut-projects/micronaut-redis/pull/244/files/6173905cb7d7b935eec544650d86feece2cdd1c7?diff=split&w=1#diff-9e98db04d68a3d2e9810c1652bd1a4a4a0c92339b293a94b3f501a417284cc5b).